--- a/arch/arm64/boot/dts/qcom/qcom-ipq6018.dtsi
+++ b/arch/arm64/boot/dts/qcom/qcom-ipq6018.dtsi
@@ -210,6 +210,16 @@
 			status = "disabled";
 		};
 
+		blsp1_uart4: serial@78b2000 {
+			compatible = "qcom,msm-uartdm-v1.4", "qcom,msm-uartdm";
+			reg = <0x78b2000 0x200>;
+			interrupts = <GIC_SPI 307 IRQ_TYPE_LEVEL_HIGH>;
+			clocks = <&gcc GCC_BLSP1_UART4_APPS_CLK>,
+				<&gcc GCC_BLSP1_AHB_CLK>;
+			clock-names = "core", "iface";
+			status = "disabled";
+		};
+
 		qcom_rng: qrng@e1000 {
 			compatible = "qcom,prng-ipq807x";
 			reg = <0xe3000 0x1000>;
